app/controllers/dams_resource_controller.rb

```diff
@@ -40,11 +40,11 @@ def show
     # generate facet collection list for collection page only
     models = @document["active_fedora_model_ssi"]
     if models.include?("DamsAssembledCollection") || models.include?("DamsProvenanceCollection") || models.include?("DamsProvenanceCollectionPart") 
-        other_rights_fquery = "(otherRights_tesim:localDisplay OR otherRights_tesim:metadataDisplay)"
-        collection_solr_params = { :q => "collections_tesim:#{params[:id]}", :fq => other_rights_fquery, :rows => 1 }
-        collection_response, collection_documents = get_search_results(collection_solr_params)
+        collection_solr_params = build_params("collections_tesim:#{params[:id]}", metadata_only_fquery)
+        collection_response = raw_solr(collection_solr_params)
         @metadata_only = !collection_response.response['numFound'].zero?
-
+        @mix_objects = mix_objects?(params[:id], collection_response.response['numFound']) if @metadata_only
```

Comment:
If `collection_response.response['numFound']` is `0` or `nil`, do we still need to call this method? It seems, after a first read, that is that is the situation. Which makes me think that `mix_objects?` is going to run the exact same query again within its method, and would therefore get the same result.

The first seems to be a few lines up w/ `collection_solr_params = build_params("collections_tesim:#{params[:id]}", metadata_only_fquery)`

The second is in the `metadata_obj_count` helper method that gets called in `mix_objects?`.

 Apologies if I read this wrong, and if so please ignore. But if `numFound` being `0` or `nil` means `mix_objects?` should ultimately return `false`, we could possibly save a query?

Reply:
@mcritchlow - I think the method mix_objects? will return false when numFound is 0 or nil for this case.  The method doesn't run the same query twice. It will only runs that query when the other place doesn't pass in that numFound like for browse by collection place  https://github.com/ucsdlib/damspas/pull/494/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR328  I can add extra line above to make @mix_objects = false by default https://github.com/ucsdlib/damspas/pull/494/files#diff-c0623f718c6b24f20f5334d0d4eaf8d5R46 What do you think?

Could you clarify more about how to save a query?  That query is only execute for browse by collection or search a collection page.  For view a single collection page, that query won't get execute twice.

Thanks

Reply:
@VivianChu - Yeah, I may likely just be misunderstanding. What I meant by save a query, was I see the first query happening here https://github.com/ucsdlib/damspas/blob/b1d806622c38c1ee6502041df03d35c61d564986/app/controllers/dams_resource_controller.rb#L43-L44

And the response `numFound` being used to pass into the `mix_objects?` method:
https://github.com/ucsdlib/damspas/blob/b1d806622c38c1ee6502041df03d35c61d564986/app/controllers/dams_resource_controller.rb#L46

 and, it seems, to call `metadata_object_count` if the response is empty https://github.com/ucsdlib/damspas/blob/b1d806622c38c1ee6502041df03d35c61d564986/lib/dams/controller_helper.rb#L932 and then the query being run there looks to me to be the same as the original https://github.com/ucsdlib/damspas/blob/b1d806622c38c1ee6502041df03d35c61d564986/lib/dams/controller_helper.rb#L912

So the 'save a query' idea would be not running that second query if the original `numFound` is `nil`, since it seemed to then do the same thing and return `false` if there's no count. But again, i may just be reading this wrong :)

Reply:
@VivianChu Maybe we can change the line 
https://github.com/ucsdlib/damspas/pull/494#discussion-diff-204544243R45 to something like `@metadata_only = collection_response.response['numFound'] > 0`, which is more easy to read, and this could be an error to avoid the nil case if it does exist?

Reply:
Oh, I'm sorry. I missed this https://github.com/ucsdlib/damspas/pull/494/files#diff-c0623f718c6b24f20f5334d0d4eaf8d5R45 where you explicitly check for this. 

Perhaps what I wonder about then is if the default `nil` value and conditional to check for it are needed in `mix_objects?`? Is there any way then for that method to be called with there _aren't_ any results?

Reply:
@mcritchlow It looks like solr will always return numFound="0" if no results are found for a query. But if the server is mis-behavior, then it should be the error handling issue.

Reply:
@lsitu - yeah, you're right. I think that assumption is perfectly reasonable. I suppose we _could_ tack on a `.to_i` to the end. But you're right I don't think that's needed at all.

It's the checking again in the `mix_objects?` method (see my edit) and the resulting second query, that I'm less clear on. Apologies @VivianChu  that I missed that guard check with `@metadata_only`